### ISSUE TITLE
Ionic Capacitor Problem

### ISSUE THREAD

**Issue by Ste-denko** [HaylLtd/cordova-plugin-background-geolocation#55]
**Describe the bug**
Trying to add this plugin to my Ionic-Angular-Capacitor project results in error when building the project:

`[ng] ./node_modules/cordova-background-geolocation-plugin/www/BackgroundGeolocation.js:13:14-40 - Error: Module not found: Error: Can't resolve 'cordova/channel' in 'my-app/node_modules/cordova-background-geolocation-plugin/ww`w'

**To Reproduce**
1. create new ionic project with Angular and Capacitor
2. add cordova-background-geolocation-plugin to project: 
`npm install cordova-background-geolocation-plugin`
3. add plugin example code to your project
4. execute `ionic serve`
5. above mentioned error pops up

**Expected behavior**
project builds without error

**Screenshots**

**Desktop (please complete the following information):**
 - OS: LinuxMint 19
 - Browser: Firefox
 - Version: latest

**Smartphone (please complete the following information):**
 - Device: MotoG7
 - OS: Android 11

**Comment by HarelM** [comment]
I'm not sure I know what the problem is but when I search google I get this issue:
https://github.com/ionic-team/capacitor/issues/1032
It might be related to the notification issue as well (see #52).
I'm not using capacitator so I'm not sure what to do here, but in case you find the solution please share it here...

**Comment by josuelmm** [comment]
Just install the packages in this way and sync, tested by me, it works perfect
```
npm i cordova-background-geolocation-plugin
npm install @awesome-cordova-plugins/background-geolocation
ionic cap sync
```

**Comment by HarelM** [comment]
Thanks @josuelmm for reporting back on this.
I'm closing since it doesn't look like we need to fix any thing...

**Comment by karimBenBechir** [comment]
Hello mates,
Issue still there, tried steps provided by @josuelmm  but still getting Can't resolve 'cordova/channel and Can't resolve 'cordova/cordova/exec
Same issues are opened on [mauron85] package and one of the merger PR is providing solution, can we have same fix here or shall we move back to the other package ? 
Thanks a lot

**Comment by HarelM** [comment]
I'm now using capacitor and this plugin, I don't have any problems.
As far as I know the other plugin isn't maintained anymore and you won't be able to use it, but you are free to choose which plugin to use.
Here's how I use it in my app, nothing special:
https://github.com/IsraelHikingMap/Site/blob/242fe282f19f42542319c36e5becf2af69087f50/IsraelHiking.Web/package.json#L45

**Comment by karimBenBechir** [comment]
Hello @HarelM, Thank you very much for your reply !
I haven't expected this quick reply !
I am using the same packages and the same enivrement's settings that you are using in israelhicking project. 
After an overall checking of the implementation, i found that everything is working fine ! 
The only imperatives to make it work are:
-  Instantiate the package like this:
import { BackgroundGeolocationPlugin} from 'cordova-background-geolocation-plugin';
declare let BackgroundGeolocation: BackgroundGeolocationPlugin;
- Test it/ use it only in capacitor enivrement: Otherwise BackgroundGeolocation will be undefined.

When trying to instantiate BackgroundGeolocation as an object of type BackgroundGeolocationPlugin, i was getting Can't resolve 'cordova/channel and Can't resolve 'cordova/exec errors on runtime and buildtime !

Thank you very much again, the package is awesome ! You rock 🤘
One last thing, do have a recommandation guideline for the battery drain and for the startOnBoot ?

**Comment by HarelM** [comment]
I'm not using start on boot. Battery drain is a problem, I believe the problem is with the screen mostly and not with the GPS from small tests we made, also the code here doesn't do much and I don't think it can be optimized much.
Having said that, there is another package from transistor soft that is more battery optimized but also less accurate, so even though I tried to use it and also paid for it, I ended up using this one (and maintain it).
